### DISCUSSION TITLE
feature: Disable Chat History

### DISCUSSION THREAD

**Discussion by StammesOpfer** [open-webui/open-webui#2892] [General]
I host a small instance with a couple shared users and auto login (single user mode). I would like to ensure that users are not seeing each others chats. Separate logins is not an option currently

I would like an option to either globally set chat history to off, or set the default behavior to off, or for chat history setting to be stored server side (it appears to be stored client side).

I am not terribly concerned about chat history getting reenabled, but an environment variable that disables chat history completely would be ideal.

**Comment by justinh-rahb** [comment]
Single user mode is exactly that, meant for a single _local_ user. Seperate logins and chat history storage is the **default** behavior that's been disabled here. I think the default setting is more appropriate for your use case. A password is not a high barrier to use, you can enter it one time and as long as you set `WEBUI_SECRET_KEY` environment variable to a static value, your session will stay logged in forever, in between container restarts or replacement.
